pkg/vault/contracts/TimelockAuthorizer.sol

```diff
@@ -240,6 +338,22 @@ contract TimelockAuthorizer is IAuthorizer, IAuthentication {
         emit ExecutionCancelled(scheduledExecutionId);
     }
 
+    /**
+     * @dev Grants or revokes permissions to grant permissions to `account` for doing `actionId` in `where`
+     * Note that pairs can revoke themselves, even revoking the root, but the root can grant himself at any time
+     */
+    function manageGrantPermission(
+        bytes32 actionId,
+        address account,
+        address where,
+        bool allowed
+    ) external {
+        bytes32 grantPermissionsActionId = getActionId(GRANT_ACTION_ID, actionId);
+        bool isAllowed = isRoot(msg.sender) || hasPermission(grantPermissionsActionId, msg.sender, where);
+        _require(isAllowed, Errors.SENDER_NOT_ALLOWED);
+        (allowed ? _grantPermission : _revokePermission)(grantPermissionsActionId, account, where);
```

Comment:
The permission to grant should indeed be `grant + action id`, but shouldn't we call `hasPermission(grant, msg.sender, where, actionId)` instead? Otherwise we would not be able to grant `grant + whatever`.

Reply:
(same with revoke)